### PR TITLE
Improve error handling / remove FIXMEs

### PR DESCRIPTION
**Pull request by maurolacy**
Improve utf-8 error handling.

Remove FIXMEs for `SliceError`s.

### CODE REVIEW

File: packages/storage-plus/src/de.rs
Comment (webmaster128) [inline packages/storage-plus/src/de.rs:107]: Arg. Here you can't do the same

```
.map_err(StdError::generic_err)?
```

because the function takes `Into<String>` instead of `ToString`?

File: packages/storage-plus/src/de.rs
Comment (ethanfrey) [inline packages/storage-plus/src/de.rs:59]: nice

File: packages/storage-plus/src/de.rs
Comment (ethanfrey) [inline packages/storage-plus/src/de.rs:137]: All these changes are the same lines:

```rust
u16::from_be_bytes(
             value
                 .as_slice()
                 .try_into()
                 .map_err(|err: TryFromSliceError| StdError::generic_err(err.to_string()))?,
         ) as usize;
```

Maybe they can be their own helper function as well, so we just call

`let t_len = parse_length(&value)?;`

**Review by ethanfrey — APPROVED** [review]
Good fix, but can you also remove those duplicates?

File: packages/storage-plus/src/de.rs
Comment (webmaster128) [reply packages/storage-plus/src/de.rs:137]: And once you do that, you can ignore the content of `err` and make it use case specific, like

```
.map_err(|_err| StdError::generic_err("Could not read 2 byte length"))?,
```

Comment (maurolacy) [reply packages/storage-plus/src/de.rs:137]: That's not easy to do, because `parse_length` would have to be generic over the integers, and I don't have an associated trait for it.

Comment (maurolacy) [reply packages/storage-plus/src/de.rs:137]: Already mentioned it in #432: https://github.com/CosmWasm/cw-plus/pull/432#discussion_r720134129

File: packages/storage-plus/src/de.rs
Comment (maurolacy) [reply packages/storage-plus/src/de.rs:107]: ✔️

File: packages/storage-plus/src/de.rs
Comment (webmaster128) [reply packages/storage-plus/src/de.rs:137]: But in this current diff it is always `u16::from_be_bytes`

Comment (maurolacy) [reply packages/storage-plus/src/de.rs:137]: Ah, you're right! Will do.

**Comment by maurolacy** [comment]
Approval was automatically dismissed or so.

**Review by ethanfrey — APPROVED** [review]
Awesome. 

BTW, at the backlog meeting during your vacation the other devs all asked for strict protected branch rules... no push on Master, no merge pr without approval and CI checks passing, dismiss stale reviews.

I applied on all rust repos I maintain. We can discuss these rules if you don't like them, happy to implement what the team requests

**Comment by maurolacy** [comment]
I agree with those rules, except perhaps dismiss stale reviews. An author can always request a new review over an already approved PR, if he deems it necessary.

**Pull request merged**
